ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/visualization-selector/visualization-selector.js

```diff
@@ -0,0 +1,207 @@
+    type: 'component',
+    componentName: 'cesium',
+    componentState: {},
+  },
+  inspector: {
+    title: 'Inspector',
+    type: 'component',
+    componentName: 'inspector',
+    componentState: {},
+  },
+  table: {
+    title: 'Table',
+    type: 'component',
+    componentName: 'table',
+    componentState: {},
+  },
+  histogram: {
+    title: 'Histogram',
+    type: 'component',
+    componentName: 'histogram',
+    componentState: {},
+  },
+}
+
+const unMaximize = contentItem => {
+  if (contentItem.isMaximised) {
+    contentItem.toggleMaximise()
+    return true
+  } else if (contentItem.contentItems.length === 0) {
+    return false
+  } else {
+    return _.some(contentItem.contentItems, subContentItem => {
+      return unMaximize(subContentItem)
+    })
+  }
+}
+
+class VisualizationSelector extends React.Component {
+  dragSources = []
+  constructor(props) {
+    super(props)
+    this.openlayers = React.createRef()
+    this.cesium = React.createRef()
+    this.inspector = React.createRef()
+    this.histogram = React.createRef()
+    this.table = React.createRef()
+  }
+  render() {
+    return (
+      <CustomElement onClick={this.handleChoice.bind(this)}>
+        <Visualization
+          ref={this.openlayers}
+          onMouseDown={this.handleMouseDown.bind(this, 'openlayers')}
+          onMouseUp={this.handleMouseUp.bind(this, 'openlayers')}
+        >
+          <VisualizationIcon className="fa fa-map" />
+          <VisualizationText>2D Map</VisualizationText>
+        </Visualization>
+        <Visualization
+          ref={this.cesium}
+          onMouseDown={this.handleMouseDown.bind(this, 'cesium')}
+          onMouseUp={this.handleMouseUp.bind(this, 'cesium')}
+        >
+          <VisualizationIcon className="fa fa-globe" />
+          <VisualizationText>3D Map</VisualizationText>
+        </Visualization>
+        <Visualization
+          ref={this.inspector}
+          onMouseDown={this.handleMouseDown.bind(this, 'inspector')}
+          onMouseUp={this.handleMouseUp.bind(this, 'inspector')}
+        >
+          <VisualizationIcon className="fa fa-info" />
+          <VisualizationText>Inspector</VisualizationText>
+        </Visualization>
+        <Visualization
+          ref={this.histogram}
+          onMouseDown={this.handleMouseDown.bind(this, 'histogram')}
+          onMouseUp={this.handleMouseUp.bind(this, 'histogram')}
+        >
+          <VisualizationIcon className="fa fa-bar-chart" />
+          <VisualizationText>Histogram</VisualizationText>
+        </Visualization>
+        <Visualization
+          ref={this.table}
+          onMouseDown={this.handleMouseDown.bind(this, 'table')}
+          onMouseUp={this.handleMouseUp.bind(this, 'table')}
+        >
+          <VisualizationIcon className="fa fa-table" />
+          <VisualizationText>Table</VisualizationText>
+        </Visualization>
+      </CustomElement>
+    )
+  }
+
+  componentDidMount() {
+    this.dragSources = []
```

Comment:
Is there a reason why this wouldn't be stored in the state?

Reply:
I think it's generally good practice to only keep variables that directly impact rendering in the state. Otherwise, you unnecessarily re-render when that variable changes